pkg/template/config_context_test.go

```diff
@@ -0,0 +1,183 @@
+package template
+
+import (
+	"testing"
+
+	kotsv1beta1 "github.com/replicatedhq/kots/kotskinds/apis/kots/v1beta1"
+	"github.com/replicatedhq/kots/kotskinds/multitype"
+	"github.com/replicatedhq/kots/pkg/crypto"
+	"github.com/stretchr/testify/require"
+	"go.undefinedlabs.com/scopeagent"
+)
+
+func TestBuilder_NewConfigContext(t *testing.T) {
+	type args struct {
+		configGroups    []kotsv1beta1.ConfigGroup
+		templateContext map[string]ItemValue
+		cipher          *crypto.AESCipher
+	}
+	tests := []struct {
+		name string
+		args args
+		want *ConfigCtx
+	}{
+		{
+			name: "empty",
+			args: args{
+				configGroups:    []kotsv1beta1.ConfigGroup{},
+				templateContext: map[string]ItemValue{},
+				cipher:          nil,
+			},
+			want: &ConfigCtx{ItemValues: map[string]ItemValue{}},
+		},
+		{
+			name: "configGroup",
+			args: args{
+				configGroups: []kotsv1beta1.ConfigGroup{
+					{
+						Name:        "abc",
+						Title:       "abc",
+						Description: "abc",
+						Items: []kotsv1beta1.ConfigItem{
+							{
+								Name:  "abcItem",
+								Type:  "text",
+								Title: "abcItem",
+								Default: multitype.BoolOrString{
+									Type:   multitype.String,
+									StrVal: "abcItemDefault",
+								},
+								Value: multitype.BoolOrString{},
+							},
+						},
+					},
+				},
+				templateContext: map[string]ItemValue{},
+				cipher:          nil,
+			},
+			want: &ConfigCtx{
+				ItemValues: map[string]ItemValue{
+					"abcItem": {
+						Value:   "",
+						Default: "abcItemDefault",
+					},
+				},
+			},
+		},
+		{
+			name: "configGroup and overriding template context value",
+			args: args{
+				configGroups: []kotsv1beta1.ConfigGroup{
+					{
+						Name:        "abc",
+						Title:       "abc",
+						Description: "abc",
+						Items: []kotsv1beta1.ConfigItem{
+							{
+								Name:  "abcItem",
+								Type:  "text",
+								Title: "abcItem",
+								Default: multitype.BoolOrString{
+									Type:   multitype.String,
+									StrVal: "abcItemDefault",
+								},
+								Value: multitype.BoolOrString{},
+							},
+						},
+					},
+				},
+				templateContext: map[string]ItemValue{
+					"abcItem": {
+						Value: "replacedAbcItemValue",
+					},
+				},
+				cipher: nil,
+			},
+			want: &ConfigCtx{
+				ItemValues: map[string]ItemValue{
+					"abcItem": {
+						Value: "replacedAbcItemValue",
+					},
+				},
+			},
+		},
+		{
+			name: "chained configOptionValue failure", // this isn't really the desired behavior, test just demonstrates it
```

Comment:
Maybe leave a little more explanation about the bug that this test with unexpected results is trying to illustrate, and/or what the expected results are meant to be...